src/txn/txn_timestamp.c

```diff
@@ -714,7 +714,22 @@ __wt_txn_set_prepare_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t prepare_
         WT_RET_MSG(session, EINVAL,
           "commit timestamp should not have been set before the prepare timestamp");
 
-    WT_RET(__txn_assert_after_reads(session, "prepare", prepare_ts));
+        /*
+         * Allow setting the prepared timestamp smaller or equal to the latest active read
+         * timestamp. This is required by MongoDB in recovery to reconstruct the prepared updates.
+         * It needs this feature to find the largest key in the table even it has been deleted. They
```

Comment:
I suggest avoiding naming MongoDB in WiredTiger code and comments. Try to write it from a general viewpoint.

Multiple ambiguous "It" in "It needs...". I suggest:
"This feature is necessary to find the largest key in the table even if that key has been deleted."

Ambiguous "They" at the end. I think it also refers to MongoDB. I don't think that sentence is helpful in WiredTiger code.